elixir/matching-brackets/lib/matching_brackets.ex

```diff
@@ -0,0 +1,22 @@
+defmodule MatchingBrackets do
+  @doc """
+  Checks that all the brackets and braces in the string are matched correctly, and nested correctly
+  """
+  @matching_brackets ["()", "[]", "{}"]
+
+  def check_brackets("") do
+    true
+  end
+
+  @spec check_brackets(String.t()) :: boolean
+  def check_brackets(str) do
+    str = Regex.replace(~r/[^(){}\[\]]/, str, "")
```

Comment:
I don't understand this regex. It seems to just handle a very specific case. Is it needed?

Reply:
Yes, it replaces everyting that is not a bracket.
In order to run my string replace function I need to get rid of all the clutter. In a string like: "\\left(\\begin{array}{cc} \\frac{1}{3} & x\\\\ \\mathrm{e}^{x} &... x^2 \\end{array}\\right)", my strategy wouldn't work.

Reply:
Oh cool! I read it wrong and thought that the ^ meant start of line for some reason!